packages/flutter/lib/src/widgets/form.dart

```diff
@@ -301,7 +305,9 @@ class FormFieldState<T> extends State<FormField<T>> {
   }
 
   bool _validate() {
-    if (widget.validator != null)
+    if (!widget.enabled)
+      _errorText = null;
```

Comment:
it seems like if it was disabled, we should do nothing, rather than clear the previous state.
on the other hand, if _validate() is being called, even if disabled, maybe we should validate regardless? and just not call it if we're disabled?

Reply:
Looking at the code, it seems like it might be better to just not autovalidate when disabled, but otherwise to continue working as now. That way if someone _wants_ to call validate() on a disabled field, it works ok.

Reply:
Won't the validation error message still appear in the UI unless you clear the state? 
Regarding the autovalidate: I thought about this but then my other though was: why would someone do validation on a disabled input anyway? And I couldn't think of a valid answer.

Reply:
It seems that the input decorator already handles this. At least I did not observe any issues during my manual tests. If you could guide me how to confirm this using unit tests, I'd be glad to write them.

Reply:
I'm not clear on what you mean by "handles this". What does it do?

The code here IMHO is definitely wrong, for the reasons given above. If we call _validate(), then we should validate. Right now we do the opposite.

Reply:
(If we are saying that this should never be called when it's disabled, then we should assert that we are enabled, not clear when we are disabled.)